Fix First Key Debounce

Chadnaut opened the pull request:
- Fix issue where the very first keypress is ignored because it fails the debounce test when `m_last_ui_cmd` in unset

Test:
- Launch (dont touch any inputs/mouse/etc)
- When loaded press TAB to enter config menu
- In current master the first keypress will be ignored

Chadnaut commented:
I can reproduce it 100% of the time.
Occurs once per load.
Run from VSCode terminal.
Fresh install, no displays.
First tab gets eaten, second tab works.
On first test `m_last_ui_cmd` is zero.

Chadnaut commented:
This snap is from master, with an extra `FeLog`
<img width="1553" height="804" alt="image" src="https://github.com/user-attachments/assets/b3dc0c31-35e9-46e4-83d1-cfec8a5ef80e" />
Commandline, Enter, Wait for load, Tab.

Chadnaut commented:
There *should* be zero existing entries on load.
How are there events before you've pressed anything?
Your `tab` works first-press because `m_last_ui_cmd` is already init by these mysterious entries.
You haven't got a gamepad with a chatty joystick connected?
A mouse on a rough surface?
Maybe Win11 is sending a focus event that Win10 does not?

Chadnaut commented:
Zero events on load should be the expected number.
What are your phantom entries that init the value?

```cpp
if ( ev.value().is<sf::Event::Closed>() ) FeLog() << "Closed" << std::endl;
if ( ev.value().is<sf::Event::Resized>() ) FeLog() << "Resized" << std::endl;
if ( ev.value().is<sf::Event::FocusLost>() ) FeLog() << "FocusLost" << std::endl;
if ( ev.value().is<sf::Event::FocusGained>() ) FeLog() << "FocusGained" << std::endl;
if ( ev.value().is<sf::Event::TextEntered>() ) FeLog() << "TextEntered" << std::endl;
if ( ev.value().is<sf::Event::KeyPressed>() ) FeLog() << "KeyPressed" << std::endl;
if ( ev.value().is<sf::Event::KeyReleased>() ) FeLog() << "KeyReleased" << std::endl;
if ( ev.value().is<sf::Event::MouseWheelScrolled>() ) FeLog() << "MouseWheelScrolled" << std::endl;
if ( ev.value().is<sf::Event::MouseButtonPressed>() ) FeLog() << "MouseButtonPressed" << std::endl;
if ( ev.value().is<sf::Event::MouseButtonReleased>() ) FeLog() << "MouseButtonReleased" << std::endl;
if ( ev.value().is<sf::Event::MouseMoved>() ) FeLog() << "MouseMoved" << std::endl;
if ( ev.value().is<sf::Event::MouseMovedRaw>() ) FeLog() << "MouseMovedRaw" << std::endl;
if ( ev.value().is<sf::Event::MouseEntered>() ) FeLog() << "MouseEntered" << std::endl;
if ( ev.value().is<sf::Event::MouseLeft>() ) FeLog() << "MouseLeft" << std::endl;
if ( ev.value().is<sf::Event::JoystickButtonPressed>() ) FeLog() << "JoystickButtonPressed" << std::endl;
if ( ev.value().is<sf::Event::JoystickButtonReleased>() ) FeLog() << "JoystickButtonReleased" << std::endl;
if ( ev.value().is<sf::Event::JoystickMoved>() ) FeLog() << "JoystickMoved" << std::endl;
if ( ev.value().is<sf::Event::JoystickConnected>() ) FeLog() << "JoystickConnected" << std::endl;
if ( ev.value().is<sf::Event::JoystickDisconnected>() ) FeLog() << "JoystickDisconnected" << std::endl;
if ( ev.value().is<sf::Event::TouchBegan>() ) FeLog() << "TouchBegan" << std::endl;
if ( ev.value().is<sf::Event::TouchMoved>() ) FeLog() << "TouchMoved" << std::endl;
if ( ev.value().is<sf::Event::TouchEnded>() ) FeLog() << "TouchEnded" << std::endl;
if ( ev.value().is<sf::Event::SensorChanged>() ) FeLog() << "SensorChanged" << std::endl;
```

Win11, it's probably `TouchBegan`

Chadnaut commented:
On Win10 I'm not getting those initial `Resized` events, which are masking the actual issue:

- `m_layout_time` was NOT getting started
- After `feOverlay.splash_logo` a `continue` gets called, skipping `feVM.tick()`
- `feVM.tick()` calls `m_layout_time.getElapsedTime()`, which starts the timer
- Since that's getting skipped, the next `FeVM::poll_command` will start the timer instead
- I have no initial `Resized` events, so my first keypress event starts it - and gets debounced because it's zero at that time.

oomek commented:
If it helps on Win10 that's fine. I haven't noticed any difference with your fix. The only 2 things I would change are to stick with `sf::Time` and rename `m_last_ui_cmd` which suggests a command not time.

Chadnaut commented:
I've reverted all previous changes and added `feVM.tick()` to the splash_logo block.
You wont see any difference because your `Resized` events kickstart the timer.
Other Windows versions might benefit. Win10 at least.
